modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php

```diff
@@ -29,96 +31,165 @@ class AZParagraphsItem extends ParagraphsItem {
+   * @param string|int $item_id
+   *   The item ID of the paragraph.
+   * @param string|int $revision_id
+   *   The revision ID of the paragraph.
+   */
+  private function processEachFieldCollection(Row $row, $field, $item_id, $revision_id) {
+    // Retrieve field collection data for the specific field.
+    $field_collection_data = $this->getFieldValues('paragraphs_item', $field, $item_id, $revision_id);
+    // Process and sort field collection values.
+    $field_collection_field_values = $this->getFieldCollectionValues($field_collection_data, $field);
+    ksort($field_collection_field_values);
+    // Set the processed values back into the row.
+    $row->setSourceProperty($field . '_values', $field_collection_field_values);
+  }
 
-      }
+  /**
+   * Retrieves and organizes field collection values.
+   *
+   * @param array $field_collection_data
+   *   The raw field collection data.
+   * @param string $field
+   *   The field collection name.
+   *
+   * @return array
+   *   The organized field collection values.
+   */
+  private function getFieldCollectionValues($field_collection_data, $field) {
+    $field_collection_field_values = [];
+    foreach ($field_collection_data as $delta => $field_collection_data_item) {
+      $field_collection_value = $this->getFieldValues(
+        'field_collection_item',
+        $field,
+        $field_collection_data_item['value'],
+        $field_collection_data_item['revision_id']
+      );
+      $field_collection_field_values[$delta] = $field_collection_value;
     }
 
+    return $field_collection_field_values;
+  }
+
+  /**
+   * Processes paragraph fields.
+   *
+   * Iterates over all fields of a paragraph and sets their values in the row.
+   *
+   * @param \Drupal\migrate\Row $row
+   *   The current row being processed.
+   * @param string|int $item_id
+   *   The item ID of the paragraph.
+   * @param string|int $revision_id
+   *   The revision ID of the paragraph.
+   */
+  private function processParagraphFields(Row $row, $item_id, $revision_id) {
     foreach (array_keys($this->getFields('paragraphs_item', $row->getSourceProperty('bundle'))) as $field) {
       $row->setSourceProperty($field, $this->getFieldValues('paragraphs_item', $field, $item_id, $revision_id));
     }
+  }
 
-    return $row;
+  /**
+   * Modifies the query to exclude archived paragraphs.
+   *
+   * This method alters the existing query by adding a condition that filters
+   * out paragraphs marked as archived. In Drupal, 'archived' often means the
+   * paragraph is either deleted or not in active use, depending on the
+   * specific implementation.
+   *
+   * By adding this condition, only active, non-archived paragraphs are fetched
+   * by the query.
+   *
+   * @param \Drupal\Core\Database\Query\SelectInterface $query
+   *   The query object that is being built for fetching paragraph items.
+   *   This object is modified by reference.
+   */
+  private function excludeArchivedParagraphs($query) {
```

Comment:
I think we may want to make this `protected` so child classes can use it.  Otherwise I think this method seems unnecessary to me.